### ISSUE TITLE
Array datasets are always written as chunked.

### ISSUE THREAD

**Issue by orenbenkiki** [scverse/Muon.jl#24]
It seem by default the Python `anndata` package writes contiguous datasets.

Perhaps an option to control this?

**Comment by orenbenkiki** [comment]
BTW, `compress=9` is deprecated (according to the latest HDF5 package I'm using, anyway). The warning I get says:

```
┌ Warning: `compress=9` keyword option is deprecated, use `deflate=9` instead
│   caller = setproperty!(p::HDF5.DatasetCreateProperties, name::Symbol, val::UInt8) at properties.jl:59
└ @ HDF5 ~/.julia/packages/HDF5/Ws1wH/src/properties.jl:59
```

**Comment by orenbenkiki** [comment]
I saw a lot of `compress=...`, mainly passed to `write` and `write_metadata`... In HDF5 the warning occurs whenever trying to set a `compress` _property_ of an object.

**Comment by orenbenkiki** [comment]
Using the head revision, I get several `WARNING: Wrapping Vararg directly in UnionAll is deprecated` when compiling Muon, but no warnings when actually using it. So the `compress` warning was due to using the ancient released version (which is what I got when using `pkg> add Muon`). I've switched the project to using the head revision (which also allows me to use the newly added `compress` flag when writing to a file - nice). I'm also voting for #25 though :-)
